packages/agent/src/commands/run.ts

```diff
@@ -0,0 +1,244 @@
+    console.log(
+      colors.gray('└── Description: ') + colors.white(agentWorkflow.description)
+    );
+
+    console.log(colors.white.bold('\nUser inputs'));
+    const inputEntries = Array.from(inputs.entries());
+    inputEntries.forEach(([key, value], idx) => {
+      const prefix = idx == inputEntries.length - 1 ? '└──' : '├──';
+      const isDefault = defaultInputs.includes(key);
+      const suffix = isDefault ? colors.gray(' (default)') : '';
+      console.log(
+        colors.white(`${prefix} ${key}=`) + colors.cyan(`${value}`) + suffix
+      );
+    });
+
+    // Validate inputs against workflow requirements
+    const validation = agentWorkflow.validateInputs(inputs);
+
+    // Display warnings if any
+    if (validation.warnings.length > 0) {
+      console.log(colors.yellow.bold('\nWarnings'));
+      validation.warnings.forEach((warning, idx) => {
+        const prefix = idx == validation.warnings.length - 1 ? '└──' : '├──';
+        console.log(colors.yellow(`${prefix} ${warning}`));
+      });
+    }
+
+    // Check for validation errors
+    if (!validation.valid) {
+      validation.errors.forEach(error => {
+        console.log(colors.red(`\n✗ ${error}`));
+      });
+      output.success = false;
+      output.error = `Input validation failed: ${validation.errors.join(', ')}`;
+    } else {
+      // Continue with workflow run
+      const stepsOutput: Map<string, Map<string, string>> = new Map();
+
+      // Print Steps
+      console.log(colors.white.bold('\nSteps'));
+      agentWorkflow.steps.forEach((step, idx) => {
+        const prefix = idx == agentWorkflow.steps.length - 1 ? '└──' : '├──';
+        console.log(
+          colors.white(`${prefix} ${idx}. `) + colors.white(`${step.name}`)
+        );
+      });
+
+      let runSteps = 0;
+
+      for (const step of agentWorkflow.steps) {
+        const runner = new Runner(
+          agentWorkflow,
+          step.id,
+          inputs,
+          stepsOutput,
+          options.agentTool,
+          options.agentModel
+        );
+
+        runSteps++;
+
+        // Run it
+        const result = await runner.run();
+
+        // Display step results
+        console.log(colors.white.bold(`\n📊 Step Results: ${step.name}`));
+        console.log(colors.gray('├── ID: ') + colors.cyan(result.id));
+        console.log(
+          colors.gray('├── Status: ') +
+            (result.success
+              ? colors.green('✓ Success')
+              : colors.red('✗ Failed'))
+        );
+        console.log(
+          colors.gray('├── Duration: ') +
+            colors.yellow(`${result.duration.toFixed(2)}s`)
+        );
+
+        if (result.tokens) {
+          console.log(
+            colors.gray('├── Tokens: ') + colors.cyan(result.tokens.toString())
+          );
+        }
+        if (result.price) {
+          console.log(
+            colors.gray('├── Price: ') +
+              colors.cyan(`$${result.price.toFixed(4)}`)
+          );
+        }
```

Comment:
I assume this is something coming from upstream, cannot help but think that 'cost' is a more accurate term though...

Reply:
Yes, cost makes more sense